CMakeLists.txt

```diff
@@ -3,7 +3,7 @@
 # accompanying file Copyright.txt for details.
 #------------------------------------------------------------------------------#
 
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
CMake 3.12 is needed in order to support compiling SST as an object library with its own dependencies and then linking the main adios2 library with the objects.  We should not merge this PR until we decide in general if bumping the requirement to CMake 3.12 is okay